File: computer-simulation-practice-code/rom.cpp
#include <iostream>
#include <string>
using namespace std;

class rom {

private:
    int* capacity;
    int size;

public:
    rom(int s) {
        size = s;
        capacity = new int[size];
}

int getSize(){
    return size;
}
 

void write(int a, int b) {
    //capacity[a] = b;
    //does nothing because it is ROM
}

int read(int a){
    cout << "I am rom          ";
    cout << capacity[a];
    return capacity[a];
}


void initialize(int adress, string a){
//starting from int adress, fill the array with each ASCII val of coorep character from a

    for(int i = 0; i < a.length(); i++){
        char myChar = a[i];
        int a = int(myChar);
        capacity[adress] = myChar;
        adress++;
        }
}

};

File: JAVA/rom.java
public class rom implements memory{
    
    private int size;
    private int[] capacity;
    
    public rom(int a){
        size = a;
        capacity = new int[size];
    
    }
    
    @Override
    public void initialize(int address, String word) {
        System.out.println("ROM initialize");

        for(int i = 0; i < word.length(); i++){
            int a = (int)word.charAt(i);
    
            capacity[address] = a;
            address++;
        }
        
    }
    
   @Override
    public int read(int address) {
        System.out.println(capacity[address]);
        return capacity[address];
    
    
    }
    
    @Override
    public void write(int address, int a) {
    // do nothing
    }

    @Override
    public int getSize() {
        return size;
    }

}
